pipeline/train/train.sh

```diff
@@ -30,42 +35,88 @@ mkdir -p "${model_dir}/tmp"
 
 all_model_metrics=(chrf ce-mean-words bleu-detok)
 
-echo "### Training ${model_dir}"
+echo "### Preparing tsv datasets and config"
+
+# Generate a new OpusTrainer config based on a template to fill paths of the datasets
+new_config="${model_dir}/config.opustrainer.yml"
+cp "configs/opustrainer/${model_type}.yml" "${new_config}"
+
+# Iterate over the training sets
+# split the input string into an array
+IFS=',' read -ra datasets <<< "${train_set_prefixes}"
+IFS=',' read -ra alns <<< "${alignments}"
+# loop through the array and get both value and index
+for index in "${!datasets[@]}"; do
+    train_set_prefix="${datasets[index]}"
+    # OpusTrainer supports only tsv and gzip
+    # TODO: pigz is not found
+    tsv_dataset="${train_set_prefix}.${src}${trg}.tsv" #.gz"
+
+    if [ "${alignments}" != "None" ] ; then
+      train_aln="${alns[index]}"
+      # alignments are currently used only for student that uses one stage of training
+      echo "### Generating tsv dataset with alignments ${alignments}"
+      paste <(${COMPRESSION_CMD} -dc "${train_set_prefix}.${src}.${ARTIFACT_EXT}") \
+            <(${COMPRESSION_CMD} -dc "${train_set_prefix}.${trg}.${ARTIFACT_EXT}") \
+            <(${COMPRESSION_CMD} -dc "${train_aln}") \
+            >"${tsv_dataset}"
+      rm "${train_aln}"
+      # when using tsv, marian requires --guided-alignments argument to be an index of the alignments in the tsv file
+      extra_params+=("--guided-alignment" "2")
+    else
```

Comment:
Thought (follow-up): This logic would be good to test when we rewrite to python. It's quite hard to follow to know the correctness and performance characteristics of this. This area _is_ hot code, since it's the raw processing of datasets.

We could test and measure this outside of CI to make sure we are using these resources correctly.